doc/fluid/design/data_pipeline/data_pipeline.md

```diff
@@ -0,0 +1,92 @@
+# Design Doc: Fluid Data Pipeline
+
+This document is about how Fluid training and inference programs reads data.
+
+## Standard Data Format
+
+### Case 1: Data from Files
+
+Consider a Fluid trianing program, `resnet50.py`, needs to read data from disk:
```

Comment:
trianing -> **training**

Reply:
Can we have a link to resnet.py.

Reply:
Done.

Reply:
Sorry I don't have a resnet.py program. It is just a placeholder here.